SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -192,7 +182,16 @@ Interest InterestFactory::CreatePlayerOwnedActorInterest() const
 
 	Query ClientQuery;
 	ClientQuery.Constraint = ClientConstraint;
-	ClientQuery.FullSnapshotResult = true;
+
+	if (GetDefault<USpatialGDKSettings>()->bEnableGranularResultTypes)
+	{
+		ClientQuery.ResultComponentId = CreateClientResultType();
```

Comment:
```suggestion
		ClientQuery.ResultComponentId = MoveTemp(CreateClientResultType());
```

Reply:
that's a fat array bruv

Reply:
only like, 1200 uints in the example project

Reply:
Perhaps I am wrong here but this shouldn't matter due to "return value optimization"?

Reply:
changed to movetemp

Reply:
As I say I don't think we need the movetemp.

Reply:
Is it harmful to have it? Whether it does anything or not, if this is the safer bet, I would rather use it

Reply:
We make this assumption in a lot of places across the gdk. I'm just trying to get Sahil to back me up @Vatyx

Reply:
We can keep it now that it's in. It does not hurt.

Reply:
Just doing a little read, apparently not using the MoveTemp (or std::move more generally) is preferable but I think it's a very small thing.

Reply:
RVO would actually be quicker - but not by much

Reply:
It does hurt, fails to build because it's const- removed again